db/memorystore.go

```diff
@@ -116,6 +119,31 @@ func (m *MemoryStore) AddAccount(acct *core.Account) (int, error) {
 	return len(m.accountsByID), nil
 }
 
+func (m *MemoryStore) ChangeAccountKey(acct *core.Account, newKey *jose.JSONWebKey) (error, *core.Account) {
+	m.Lock()
+	defer m.Unlock()
+
+	oldKeyID, err := keyToID(acct.Key)
+	if err != nil {
+		return err, nil
+	}
+
+	newKeyID, err := keyToID(newKey)
+	if err != nil {
+		return err, nil
+	}
+
+	if otherAccount, present := m.accountsByKeyID[newKeyID]; present {
+		return fmt.Errorf("New public key is already in use"), otherAccount
+	}
+
+	delete(m.accountsByKeyID, oldKeyID)
+	acct.Key = newKey
+	m.accountsByKeyID[newKeyID] = acct
+	m.accountsByID[acct.ID] = acct
+	return nil, nil
```

Comment:
I think this should be just `return nil` once the refactoring I describe happens.